### PR TITLE
Feature/1119 ssp section 3

### LINKED ISSUE #1119: SSP Aggregate (03) File Parsing - As a data analyst, I want my data files to be parsed so that my submission can be validated/accepted by TDP
Description:
Parsing of SPP STT submissions for Aggregate Data.

AC: 
- [ ] Data File parsing logic is in place such that data file submission can be completed by user and parsed by system for full submission once errors are resolved.
- [ ] Data Errors tell user specifically where in the file they occur, and therefore what exactly needs to be fixed in order for the submission to be complete
- [ ] TDP gives the user all data errors for a submission in a way they can easily save or export so they can take the issues back to their case management software and fix the issues then recreate an export for resubmission to TDP.
- [ ] Data Errors must be issued in human readable plain language so that an english speaker can easily understand what is needed. 

Tasks: 
- [ ] Sample tasks

DoD: 
 - [ ] minimum 90% code coverage test
 - [ ] 0 linting errors and warnings
 - [ ] 0 accessibility errors
 - [ ] deployed to a dev environment
 - [ ] all significant changes have been documented
 - [ ] free of high and medium security vulnerabilities

Supporting Documentation:
- [_SSP_Validation_Workbook_03_](https://gorafttech.sharepoint.com/:x:/r/sites/TDRSResearchDesign/_layouts/15/Doc.aspx?sourcedoc=%7B2A8ADA1D-EE54-4C52-BAC3-1610CDC8C3BA%7D&file=TDP_Validation_SSP_Section3.xlsx&action=default&mobileredirect=true)
- [_Parsing_Notebook](https://gorafttech.sharepoint.com/sites/TDRSResearchDesign/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FTDRSResearchDesign%2FShared%20Documents%2FDev%2Fdev%2FParsing%2Fhtml&viewid=981949f0%2D98e1%2D4ace%2D8da8%2D436aca592a46&noAuthRedirect=1)
- [_SSP_Instructions_Doc_](https://gorafttech.sharepoint.com/:w:/r/sites/TDRSResearchDesign/_layouts/15/Doc.aspx?sourcedoc=%7BFD51E66D-6DDA-47AA-A970-29006EEC07F3%7D&file=TANF-SSP%20data%20instructions_clean.docx&action=default&mobileredirect=true)

Open Questions:

(any clarifying questions in support of the issue)

### PR DESCRIPTION
**Pull request by jtimpe**
## Summary of Changes
Pull request closes #1119

## How to Test
```
cd tdrs-backend && docker-compose up
cd tdrs-frontend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
1. Upload the data file `ADS.E2J.NDM3.MS24`, selecting SSP-MOE as the program type
1. Ensure the resulting submission history statuses and error files match the expected output

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Data File parsing logic is in place such that data file submission can be completed by user and parsed by system for full submission once errors are resolved.
+ [x] Data Errors tell user specifically where in the file they occur, and therefore what exactly needs to be fixed in order for the submission to be complete
+ [x] TDP gives the user all data errors for a submission in a way they can easily save or export so they can take the issues back to their case management software and fix the issues then recreate an export for resubmission to TDP.
+ [x] Data Errors must be issued in human readable plain language so that an english speaker can easily understand what is needed.
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/parsers/schema_defs/ssp/m6.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        Field(item="4C", name='NUM_2_PARENTS', type='number', startIndex=47, endIndex=55,
+              required=True, validators=[validators.isInLimits(0, 99999999)]),
+        Field(item="5C", name='NUM_1_PARENTS', type='number', startIndex=71, endIndex=79,
+              required=True, validators=[validators.isInLimits(0, 99999999)]),
+        Field(item="6C", name='NUM_NO_PARENTS', type='number', startIndex=95, endIndex=103,
+              required=True, validators=[validators.isInLimits(0, 99999999)]),
+        Field(item="7C", name='NUM_RECIPIENTS', type='number', startIndex=119, endIndex=127,
+              required=True, validators=[validators.isInLimits(0, 99999999)]),
+        Field(item="8C", name='ADULT_RECIPIENTS', type='number', startIndex=143, endIndex=151,
+              required=True, validators=[validators.isInLimits(0, 99999999)]),
```
Comment (elipe17) [inline tdrs-backend/tdpservice/parsers/schema_defs/ssp/m6.py:128]: Is this supposed to have a validator or did Alex make a mistake in the spreadsheet?

**Review by elipe17 — APPROVED** [review]

**Review by andrew-jameson — APPROVED** [review]

**Review by George-Hudson — APPROVED** [review]
LGTM

**Review by raftmsohani — APPROVED** [review]
LGTM!

**Review by ADPennington — APPROVED** [review]
looks great @jtimpe 🚀 

- checked the clean files as well as those with summing issues. 
- also checked that the transformation from qtr to report month was accurate
ssp_agg_qasp

**Pull request merged**